wgengine/magicsock/magicsock_linux.go

```diff
@@ -5,28 +5,36 @@ package magicsock
 
 import (
 	"bytes"
-	"encoding/binary"
+	"context"
 	"errors"
 	"fmt"
 	"io"
 	"net"
 	"net/netip"
 	"syscall"
 	"time"
-	"unsafe"
 
+	"github.com/mdlayher/socket"
```

Comment:
I'm glad this packed ended up working so well!

Reply:
Yeah, huge improvement over hand-rolling this, IMO.